src/addon/file_server/mod.rs

```diff
@@ -196,11 +197,11 @@ impl<'a> FileServer {
     /// This happens because links should behave relative to the `/` path
     /// which in this case is `http-server/src` instead of system's root path.
     fn make_dir_entry_link(root_dir: &Path, entry_path: &Path) -> String {
-        // format!("/{}", &entry_path[current_dir_path.len()..])
         let root_dir = root_dir.to_str().unwrap();
         let entry_path = entry_path.to_str().unwrap();
+        let path_buf = PathBuf::from_str(&entry_path[root_dir.len()..]).unwrap();
 
-        entry_path[root_dir.len()..].to_string()
+        encode_uri(&path_buf)
     }
```

Comment:
Here's the code that turns the file path into a valid URI string. This is then passed to the File Explorer to provide links to navigate.